src/utils/transformAwareTranslator.ts

```diff
@@ -0,0 +1,96 @@
+import * as d3 from "d3";
+
+import * as Utils from "../utils";
+
+import { Point } from "../";
+import { SimpleSelection } from "../core/interfaces";
+import { Component } from "../components/component";
+
+const _TRANSLATOR_KEY = "__Plottable_ClientTranslator";
+
+export function getTranslator(component: Component): Translator {
+  // The Translator works by first calculating the offset to root of the chart and then calculating
+  // the offset from the component to the root. It is imperative that the measureElement
+  // be added to the root of the hierarchy and nowhere else.
+  let root = component.root().rootElement().node() as Element;
+
+  let translator: Translator = (<any> root)[_TRANSLATOR_KEY];
+  if (translator == null) {
+    const measurer = <SVGElement> <any>document.createElementNS(root.namespaceURI, "svg");
+
+    measurer.setAttribute("class", "measurer");
+    measurer.setAttribute("style", "opacity: 0; visibility: hidden; position: absolute; width: 1px; height: 1px;");
+
+    root.appendChild(measurer);
+
+    translator = new Translator(d3.select(measurer));
+    (<any> root)[_TRANSLATOR_KEY] = translator;
+  }
+
+  return translator;
+}
+
+/**
+ * Applies position as a style and attribute to the svg element
+ * as the position of the element varies by the type of parent.
+ * When nested within an SVG, the attribute position is respected.
+ * When nested within an HTML, the style position is respected.
+ */
```

Comment:
I actually did some poking around and I think we can get rid of this whole thing by using `event.offsetX`/`event.offsetY`: https://jsfiddle.net/hellochar/3soLpj9f/